operators/slang_shader/slang_shader_op.cpp

```diff
@@ -0,0 +1,344 @@
+    SLANG_DIAGNOSE_IF_NEEDED(diagnostics_blob);
+    if (!holoscan_module_) {
+      throw std::runtime_error("Failed to compile Slang module");
+    }
+  }
+
+  Slang::ComPtr<slang::ISession> session_;
+
+  std::unique_ptr<SlangShader> shader_;
+
+  std::list<std::unique_ptr<Command>> pre_launch_commands_;
+  std::list<std::unique_ptr<Command>> launch_commands_;
+  std::list<std::unique_ptr<Command>> post_launch_commands_;
+
+ private:
+  // Slang compilation components
+  Slang::ComPtr<slang::IGlobalSession> global_session_;
+  Slang::ComPtr<slang::IModule> holoscan_module_;
+};
+
+void SlangShaderOp::initialize() {
+  // Add the allocator to the operator so that it is initialized, this is needed because the
+  // parameter has a default allocator
+  add_arg(allocator_.default_value());
+
+  // Call the base class initialize function
+  Operator::initialize();
+}
+
+void SlangShaderOp::setup(OperatorSpec& spec) {
+  assert(!impl_);
+  impl_ = std::make_shared<Impl>();
+
+  spec.param(shader_source_, "shader_source", "Shader source string.", "Shader source string.");
+
+  spec.param(
+      shader_source_file_, "shader_source_file", "Shader source file.", "Shader source file.");
+
+  spec.param(allocator_,
+             "allocator",
+             "Allocator for output buffers.",
+             "Allocator for output buffers.",
+             std::static_pointer_cast<Allocator>(
+                 fragment()->make_resource<UnboundedAllocator>("allocator")));
+
+  // Add a CUDA stream pool
+  add_arg(fragment()->make_resource<CudaStreamPool>("cuda_stream", 0, 0, 0, 1, 0));
+
+  // We need the shader source to build the input and output ports and the parameters, so check
+  // the argument list and get them
+  std::string shader_source, shader_source_file;
+  for (auto&& arg : args()) {
+    if (arg.name() == "shader_source") {
+      shader_source = std::any_cast<std::string>(arg.value());
+    } else if (arg.name() == "shader_source_file") {
+      shader_source_file = std::any_cast<std::string>(arg.value());
+    }
+  }
+
+  if (shader_source.empty() && shader_source_file.empty()) {
+    throw std::runtime_error("Either 'shader_source' or 'shader_source_file' must be set");
+  }
+
+  if (!shader_source.empty() && !shader_source_file.empty()) {
+    throw std::runtime_error("Both 'shader_source' and 'shader_source_file' cannot be set");
+  }
+
+  // load from file
+  if (!shader_source_file.empty()) {
+    std::ifstream in_stream(shader_source_file);
+    if (!in_stream.is_open()) {
+      throw std::runtime_error(fmt::format("Failed to open shader file '{}'", shader_source_file));
+    }
+    std::stringstream shader_string;
+    shader_string << in_stream.rdbuf();
+    shader_source = shader_string.str();
+  }
+
+  impl_->shader_ = std::make_unique<SlangShader>(impl_->session_, shader_source);
+
+  const nlohmann::json reflection = impl_->shader_->get_reflection();
+
+  // Get Slang parameters and setup input, outputs and parameters
```

Comment:
multiple inputs are suppored?
```suggestion
  // Get Slang parameters and setup inputs, outputs and parameters
```